examples/raft-kv-memstore/src/network/raft_network_impl.rs

```diff
@@ -53,12 +53,12 @@ impl ExampleNetwork {
 impl RaftNetworkFactory<ExampleTypeConfig> for ExampleNetwork {
     type Network = ExampleNetworkConnection;
 
-    async fn connect(&mut self, target: ExampleNodeId, node: Option<&Node>) -> Self::Network {
-        ExampleNetworkConnection {
+    async fn connect(&mut self, target: ExampleNodeId, node: Option<&Node>) -> Result<Self::Network, NetworkError> {
+        Ok(ExampleNetworkConnection {
```

Comment:
It's better to have another associated type like `type Network` to define an application network error so that an application can define an Error type it wants to use.

And when such an error occurs, it's openraft's responsibility to convert it to `openraft::NetworkError` if needed.